pkg/phlaredb/block_querier.go

```diff
@@ -1216,15 +1216,26 @@ func (r *inMemoryparquetReader[M, P]) open(ctx context.Context, bucketReader phl
 	}
 
 	// read all rows into memory
-	r.cache = make([]M, r.file.NumRows())
+	r.cache = make([]M, 0, r.file.NumRows())
 	var read int
 	for _, rg := range r.file.RowGroups() {
-		err := func(rg parquet.RowGroup) error {
-			reader := parquet.NewGenericRowGroupReader[M](rg)
-			defer runutil.CloseWithLogOnErr(util.Logger, reader, "closing parquet generic row group reader")
+		err = func(rg parquet.RowGroup) error {
```

Comment:
I suspect we should also run this in parallel ?

Reply:
I'm not sure. The problem is that it would cost us lots of memory, because [page reader acqiures a buffer](https://github.com/segmentio/parquet-go/blob/main/file.go#L519) (of `ReadBufferSize` size 2MB) per column chunk, and being read concurrently, the internal buffer pool would be useless.

Interestingly, I see no apparent reason why column chunks should not be read concurrently: https://github.com/segmentio/parquet-go/blob/5d42db8f0d4728c31759068f08da15df44c6cc7f/row_group.go#L320, since buffers are already allocated, it should not be this wasteful

Reply:
yeah column are read in parallel in that case, but I thought it would be good to process multiple rowgroup at once, not sure if it would help as you say let's see.